### PR TITLE
[T3A3][W09-A1] Low Yong Siang

### PR DESCRIPTION
**Pull request by YongSiang94**
Ready for review

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -45,8 +45,8 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+* `add John Doe p/98765432 e/johnd@gmail.com a/block 123, John street, #01-01, 123456`
+* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/234, Newgate Prison, #01-02, 234567 t/criminal t/friend`
```
Comment (jeffryhartanto) [inline doc/UserGuide.md]: Good job for editing the UserGuide.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,12 +8,24 @@
...
-    public final String value;
+    public static final String EXAMPLE = "123, some street, some unit, some postal code";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in: a/BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+,.+,.+,.+";
+    public static final String COMMA_SEPARATOR = ",";
+    
+    private static final int BLOCK_INDEX = 0;
+    private static final int STREET_INDEX = 1;
+    private static final int UNIT_INDEX = 2;
+    private static final int POSTAL_INDEX = 3;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: How about using enum?

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -8,12 +8,24 @@
...
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
+    public static final String EXAMPLE = "123, some street, some unit, some postal code";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in: a/BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+,.+,.+,.+";
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: Good job for using regex.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -26,9 +38,20 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        // this.value = trimmedAddress;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -26,9 +38,20 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+    /**
+     * Creates multiple smaller class (Block, Street, Unit, PostalCode) for the Address class.
+     */
+    public void initiatesAddrSubClass() { 
+    	this.block = new Block(splitAddress[BLOCK_INDEX]);
+    	this.street = new Street(splitAddress[STREET_INDEX]);
+    	this.unit = new Unit(splitAddress[UNIT_INDEX]);
+    	this.postal = new PostalCode(splitAddress[POSTAL_INDEX]);
+    }
+
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: When do you call this method?

File: src/seedu/addressbook/data/person/Block.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Block.java]: Don't forget to add header comment for class.

File: src/seedu/addressbook/data/person/Block.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	
+	private String blockNumber;
+
+	/**
+	 * Creates a Block object with the number given
+	 * @param number
+	 */
+	public Block(String number) {
+		this.setBlockNumber(number);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Block.java]: Please fix the indentation and change the tab to spaces in Eclipse, see [here](http://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)

File: src/seedu/addressbook/data/person/Unit.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	/**
+	 * 
+	 * @return the unitNumber of that unit.
+	 */
+	public String getUnitNumber() {
+		return unitNumber;
+	}
+
+	/**
+	 * Sets the new unitNumber for that unit.
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Unit.java]: `Changes` or `Sets` ? be consistent. :)

File: src/seedu/addressbook/data/person/Unit.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+	 * Creates a new Unit object with the unitNumber given
+	 * @param number
+	 */
+	public Unit(String number) { 
+		this.setUnitNumber(number);
+	}
+
+	/**
+	 * 
+	 * @return the unitNumber of that unit.
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Unit.java]: Better write method description than `@return`. Refer to #188

**Comment by jeffryhartanto** [comment]
@YongSiang94 Some comments added. Please close the PR after reading comments.

When you modify the behavior, remember to update tests as well.

**Comment by YongSiang94** [comment]
Alright! Thank you!

File: src/seedu/addressbook/data/person/Address.java (hunk 3 above)
Comment (funnieguy) [inline src/seedu/addressbook/data/person/Address.java]: This would match "123, some street, some unit, some postal code, <something extra>"

**Pull request closed without merging**